detection-rules/credential_phishing_suspicious_file_share_link_or_html.yml

```
@@ -0,0 +1,62 @@
+name: "Suspicious file sharing via link or HTML attachment"
+description: "This rule detects messages with verbiage around sharing a file, this has been observed abusing links and HTML documents. Links are further analyzed for credential phishing or suspicious tld indicators. Requires a new or unsolicited sender."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and length(body.current_thread.text) < 800
+  and regex.icontains(body.current_thread.text,
+                      '(shared.{0,30}with you|View Document)'
+  )
+  and (
+    any(body.links,
+        (
+          beta.linkanalysis(.).credphish.contains_login
+          or beta.linkanalysis(.).credphish.contains_captcha
+        )
+  
+        // or linkanalysis concludes phishing of medium to high confidence
+        or any([beta.linkanalysis(.)],
+               (
+                 .credphish.disposition == "phishing"
+                 and .credphish.brand.confidence in ("medium", "high")
+               )
+               or any(file.explode(.screenshot),
+                      strings.contains(.scan.ocr.raw, "link has been removed")
+               )
+        )
+        or .href_url.domain.tld in $suspicious_tlds
```

```suggestion
        or (.href_url.domain.tld in $suspicious_tlds and sender.email.domain.tld not in $suspicious_tlds)
```
defer to you if this is worth it